libs/internal/include/launchdarkly/logging/spy_logger.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <launchdarkly/logging/log_backend.hpp>
+#include <launchdarkly/logging/logger.hpp>
+
+#include <vector>
+
+namespace launchdarkly::logging {
+/**
+ * Creates a logger for tests where log messages need to be inspected.
+ *
+ * @return Logger suitable for passing to functions accepting a
+ * launchdarkly::Logger.
+ */
+launchdarkly::Logger SpyLogger(std::vector<std::string>& messages);
+
+class SpyLoggerBackend : public launchdarkly::ILogBackend {
+   public:
+    SpyLoggerBackend(std::vector<std::string>& messages);
```

Comment:
I think the vector should be tuples containing the level and the string value. So you can validate not just the log message, but that it was logged as the correct level.

suggestion: A helper for asserting on one or more expected messages lowers the barrier of entry quite a bit as well. 
Example from js-core to illustrate.
```node
      logger.expectMessages([
        {
          level: LogLevel.Error,
          matches: /Received error.*giving up permanently/,
        },
```

Reply:
That's fair. I'll add that.

Reply:
Added this API using `:::testing::AssertionResult` helpers:
```c++
    ASSERT_TRUE(messages_->Count(1));
    ASSERT_TRUE(messages_->Contains(0, LogLevel::kError, "some string"));
```
Can extend if we need more matchers.

Reply:
I like it.